awx/main/queue.py

```diff
@@ -30,6 +30,9 @@ def __init__(self):
         self.queue = getattr(settings, 'CALLBACK_QUEUE', '')
         self.logger = logging.getLogger('awx.main.queue.CallbackQueueDispatcher')
         self.connection = redis.Redis.from_url(settings.BROKER_URL)
+        self.subsystem_metrics = s_metrics.Metrics(conn=self.connection)
+        self.connection.client_setname("callbackqueuedispatcher")
 
     def dispatch(self, obj):
         self.connection.rpush(self.queue, json.dumps(obj, cls=AnsibleJSONEncoder))
+        self.subsystem_metrics.inc('callback_receiver_events_insert_redis', 1)
```

Comment:
In my testing, this per-event inc *really* seems to drastically slow down event insertion.

Reply:
We might consider just getting rid of this metric; it seems really close to `llen(callback_tasks)`, and doing this additional `inc()` per-write is just plain expensive.